tests/js/common/shell/shell-pregel-start-cluster.js

```diff
@@ -134,30 +135,21 @@ function basicTestSuite() {
   function testAlgo(a, p) {
     p.shardKeyAttribute = shardKey;
     var pid = pregel.start(a, graphName, p);
-    var i = 10000;
-    do {
-      internal.wait(0.2);
-      var stats = pregel.status(pid);
-      if (stats.state !== "loading" && stats.state !== "running" && stats.state !== "storing") {
-        assertEqual(stats.vertexCount, 11, stats);
-        assertEqual(stats.edgeCount, 17, stats);
+    const stats = pregelTestHelpers.waitUntilRunFinishedSuccessfully(pid);
+
+    assertEqual(stats.vertexCount, 11, stats);
+    assertEqual(stats.edgeCount, 17, stats);
   
-        db[vColl].all().toArray()
-          .forEach(function (d) {
-            if (d[a] && d[a] !== -1) {
-              var diff = Math.abs(d[a] - d.result);
-              if (diff > EPS) {
-                console.log("Error on " + JSON.stringify(d));
-                assertTrue(false);// peng
-              }
-            }
-          });
-        break;
-      }
-    } while (i-- >= 0);
-    if (i === 0) {
-      assertTrue(false, "timeout in pregel execution");
-    }
+    db[vColl].all().toArray()
+      .forEach(function (d) {
+        if (d[a] && d[a] !== -1) {
+          var diff = Math.abs(d[a] - d.result);
+          if (diff > EPS) {
+            console.log("Error on " + JSON.stringify(d));
+            assertTrue(false);// peng
```

Comment:
I think
```suggestion
            assertTrue(false, "Error on " + JSON.stringify(d));// peng
```
might be more helpful as failure output?